docs/en/observability/monitor-k8s/monitor-k8s-application-performance.asciidoc

```diff
@@ -2,52 +2,220 @@
+
+* {apm-go-ref-v}/getting-started.html[Go agent]
+* {apm-java-ref-v}/setup.html[Java agent]
+* {apm-dotnet-ref-v}/setup.html[.NET agent]
+* {apm-node-ref-v}/set-up.html[Node.js agent]
+* {apm-py-ref-v}/set-up.html[Python agent]
+* {apm-ruby-ref-v}/set-up.html[Ruby agent]
+
+[discrete]
+=== Step 3: (Optional) Attach the agent
+
+Some agents offer the ability to instrument supported technologies without
+any changes to an application image or code.
+To do this, you'll need an https://kubernetes.io/docs/concepts/workloads/pods/init-containers/[init container]
+based on an official Elastic APM docker image.
+
+Before your application starts, copy the agent from the init container into a shared volume.
+For example, with the Java agent:
+
+[source,yml]
+----
+    # ...
+    spec:
+      volumes:
+      - name: elastic-apm-agent <1>
+        emptyDir: {}
+      initContainers:
+      - name: elastic-java-agent
+        image: docker.elastic.co/observability/apm-agent-java:1.12.0 <2>
+        volumeMounts:
+        - mountPath: /elastic/apm/agent
+          name: elastic-apm-agent
+        command: ['cp', '-v', '/usr/agent/elastic-apm-agent.jar', '/elastic/apm/agent'] <3>
+----
+<1> The shared volume
+<2> Build the initContainer from the official Elastic Java agent image
+<3> Copy the agent to the shared volume
+
+The Java command line needs a way to pick up this `javaagent` configuration.
+You can use the standard JVMTI https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#tooloptions[JAVA_TOOL_OPTIONS] environment variable to do this.
+It doesn't have to be explicitly specified and is picked up automatically by the JVM when it starts.
+
+TIP: For JVMs that don't support this option, you can use any other environment variable --
+either one already defined in your startup script, like `JAVA_OPTS` in some servlet container scripts,
+or add a dedicated empty one that will have no effect if it's not set.
+
+[source,yml]
+----
+      # ...
+      containers:
+      - name: your-app-container
+        env:
+        # ...
+        - name: JAVA_TOOL_OPTIONS <1>
+          value: -javaagent:/elastic/apm/agent/elastic-apm-agent.jar
+----
+<1> Used for the command line to pick up the `javaagent` configuration
 
 [discrete]
-==== Download and install
+=== Step 4: Save the secret token
+
+A {apm-server-ref-v}/secret-token.html[secret token] is used to secure communication between APM agents
+and APM Server. In your {ecloud} deployment page, select *APM* and copy your APM Server secret token.
+If you're managing APM Server yourself, see {apm-server-ref-v}/secret-token.html[secret token] for instructions
+on how to set up your secret token.
 
-TODO: Describe how to download and install APM server from archives.
+To avoid exposing the secret token, you can store it in a Kubernetes secret. For example:
+
+[source, cmd]
+----
+kubectl create secret generic apm-secret --from-literal=ELASTIC_APM_SECRET_TOKEN=asecretpassword --namespace=kube-system <1>
```

Comment:
noting here that I didn't test the secret store. I hard coded the secret to get things working because I had other problems, and never got back to testing this step. Sorry. :-(

Reply:
That's ok. I rewrote these to match your secret commands for logs and metrics.